src/jit/phase.h

```diff
@@ -6,77 +6,99 @@
 #ifndef _PHASE_H_
 #define _PHASE_H_
 
+// Specify which checks a phase should run
+//
+enum class PhaseChecks
+{
+    NONE,
+    ALL
+};
+
+// A phase encapsulates a part of the compilation pipeline for a method.
+//
 class Phase
 {
 public:
     virtual void Run();
 
 protected:
-    Phase(Compiler* _comp, const char* _name, Phases _phase = PHASE_NUMBER_OF)
-        : comp(_comp), name(_name), phase(_phase), doChecks(true)
+    Phase(Compiler* _compiler, Phases _phase, PhaseChecks _checks = PhaseChecks::ALL)
+        : comp(_compiler), m_name(nullptr), m_phase(_phase), m_checks(_checks)
     {
+        m_name = PhaseNames[_phase];
     }
 
     virtual void PrePhase();
     virtual void DoPhase() = 0;
     virtual void PostPhase();
 
     Compiler*   comp;
-    const char* name;
-    Phases      phase;
-    bool        doChecks;
+    const char* m_name;
+    Phases      m_phase;
+    PhaseChecks m_checks;
 };
 
-inline void Phase::Run()
-{
-    PrePhase();
-    DoPhase();
-    PostPhase();
-}
-
-inline void Phase::PrePhase()
+// A phase that accepts a lambda for the actions done by the phase.
+//
+// Would prefer to use std::function via <functional> here, but seemingly can't.
```

Comment:
Why  can't  `std::function<void()>` be used here?

Reply:
#include of <functional> causes build errors. I did not drill into why.

Reply:
@sandreenko Presumably you're referring to the real STL/std, right? Unfortunately it can't be used in the JIT due to conflicts with utilcode, pal etc. That's why there's `jitstd` (which of course, doesn't have `function`).

Reply:
Yeah, std.. Thank you.